### PR TITLE
Improve `leancode_lint` readme

### PR DESCRIPTION
**Pull request by bartekpacia**
I found it a bit unclear, so googled a bit and added an explanation that would have satisfied me and not require any googling.

### CODE REVIEW

File: packages/leancode_lint/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# 1.0.2
```
Comment (shilangyu) [inline packages/leancode_lint/CHANGELOG.md:1]: Should be a `1.0.1+1` imo since it adds nothing to the package itself

Comment (bartekpacia) [reply packages/leancode_lint/CHANGELOG.md:1]: That's a good point, I haven't thought of it.

File: packages/leancode_lint/README.md [hunk 2]
```diff
@@ -6,11 +6,19 @@ Lint rules used internally in LeanCode projects.
...
 
-Add `leancode_lint` as a normal dependency to access the `unawaited` function when using dart version lower than 2.14.
+Add `leancode_lint` as a dev dependency.
+
+```yaml
+dev_dependencies:
+  leancode_lint: ^1.0.1+1
+```
+
+When using Dart < 2.14, consider instead adding `leancode_lint` as a normal dependency
```
Comment (Albert221) [inline packages/leancode_lint/README.md:16]:
```suggestion
When using Dart <2.14, consider adding `leancode_lint` as a normal dependency instead
```

File: packages/leancode_lint/README.md [hunk 3]
```diff
@@ -6,11 +6,19 @@ Lint rules used internally in LeanCode projects.
 
 ## Usage
 
-Add `leancode_lint` as a normal dependency to access the `unawaited` function when using dart version lower than 2.14.
+Add `leancode_lint` as a dev dependency.
+
+```yaml
```
Comment (Albert221) [inline packages/leancode_lint/README.md:11]: I'd put this snippet and the paragraph below under a h3 `Dart <2.14`

Comment (bartekpacia) [reply packages/leancode_lint/README.md:11]: done

File: packages/leancode_lint/README.md (hunk 2 above)
Comment (bartekpacia) [reply packages/leancode_lint/README.md]: thanks

**Review by Albert221 — APPROVED** [review]

**Review by RobertOdrowazLNCD — APPROVED** [review]

**Pull request merged**